internal/storage/persistence/migrations/20240507234842_location_source_amss.down.sql

```diff
@@ -0,0 +1,2 @@
+-- reverse: modify "location" table
+ALTER TABLE `location` MODIFY COLUMN `source` enum('unspecified','minio','sftp') NOT NULL;
```

Comment:
I think the db migrations would benefit from a second look and maybe some changes.  I've never fully understood how the ent/atlas db migration system works, but it looks to me like https://github.com/artefactual-sdps/enduro/blob/main/cmd/migrate/main.go is pretty close to the example provide by ent at https://entgo.io/docs/versioned-migrations.

I've been reading the ent and Atlas docs, but I can't figure out if Atlas generates "down" migration file by default, or if this is something that ent migrations do by default.  Looking at the code I don't see any flags that we are using that would enable generation of down migration files. :confused:

Reply:
https://atlasgo.io/blog/2024/04/01/migrate-down is an interesting read.  My main takeaway is that the atlas down migrations are mainly intended for use in development, and not in production due to their destructive nature.

Reply:
I see a few differences between this migration script and the ones we have in other projects (where the down migrations are not created). Most important ones, the usage of `sqltool.NewGolangMigrateDir(path)` instead of `atlas.NewLocalDir(migrationsDir)` and the addition of `schema.WithFormatter(atlas.DefaultFormatter)` to the migrate options, maybe those have something to do. Not a big deal in any case.